src/opendr/perception/facial_expression_recognition/image_based_facial_emotion_estimation/facial_emotion_learner.py

```diff
@@ -0,0 +1,804 @@
+        """
+        This method loads an ONNX model from the path provided into an onnxruntime inference session.
+        :param path: path to ONNX model
+        :type path: str
+        """
+        self.ort_session = onnxruntime.InferenceSession(path)
+
+    def reset(self):
+        """This method is not used in this implementation."""
+        return NotImplementedError
+
+    def download(self, path=None, mode="data", url=OPENDR_SERVER_URL + "perception/facial_emotion_estimation"):
+        """
+        This method downloads data files and saves them in the path provided.
+        :param path: Local path to save the files, defaults to self.temp_path if None
+        :type path: str, path, optional
+        :param mode: Whether to download data or the pretrained model, or image/video for running demo
+        :type mode: It can be an item in ["data", "pretrained", "demo_image", "demo_video"]
+        :param url: URL of the FTP server, defaults to OpenDR FTP URL
+        :type url: str, optional
+        """
+
+        valid_modes = ["data", "pretrained", "demo_image", "demo_video"]
+        if mode not in valid_modes:
+            raise UserWarning("mode parameter not valid:", mode, ", file should be one of:", valid_modes)
+        if path is None:
+            path = self.temp_path
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        if mode == "data":
+            print("Downloading data...")
+            data_path = os.path.join(path, 'data')
+            if not os.path.exists(data_path):
+                os.makedirs(data_path)
+
+            zip_path = os.path.join(path, 'data/AffectNet_micro.zip')
+            if not os.path.exists(zip_path):
+                # Download data
+                file_url = os.path.join(url, 'data/AffectNet_micro.zip')
+                urlretrieve(file_url, zip_path)
+                with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+                    zip_ref.extractall(data_path)
+                os.remove(zip_path)
+            else:
+                print("Data files already exist.")
+            print("Data download complete.")
+            downloaded_files_path = os.path.join(data_path, 'AffectNet_micro')
+
+        elif mode == "pretrained":
+            print("Downloading pretrained model weights...")
+            model_path = os.path.join(path, 'pretrained')
+            if not os.path.exists(model_path):
+                os.makedirs(model_path)
+
+            zip_path = os.path.join(path, 'pretrained/esr_9.zip')
+            if not os.path.exists(zip_path):
+                # Download data
+                file_url = os.path.join(url, 'pretrained/esr_9.zip')
+                urlretrieve(file_url, zip_path)
+                with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+                    zip_ref.extractall(model_path)
+                os.remove(zip_path)
+            else:
+                print("Pretrained files already exist.")
```

Comment:
I noticed that this code ends up re-downloading every time. The exists check is made for the zip file which is removed on line 766. It should probably check for the extracted directory or maybe it shouldn't remove the zip file, whichever you prefer.

Same applies for data files on line 741,etc. above.

Reply:
Thank you for the comment. I removed the zip remover.